### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by ErlingHauan**
## Summary
Limits access to Dashboard and FormBuilder to logged in users.

## Backend
* `/api/users/login` now creates the `authorizedUser` cookie that contains the user's hashed email, and transmits it to the browser.
* Added the `/api/users/verify` endpoint, which checks if the `authorizedUser` browser cookie is present.
* Added the `/api/users/logout` endpoint, which removes the `authorizedUser` cookie. Due to pending design changes, a `Log out` button has not been implemented in this PR. A separate issue has been created: #139.

## Frontend
* Added a custom `useAuthorization` hook, which verifies the user's session via an API call. If not verified, it redirects to the login page.
* After redirecting, the login page displays an error.
* Removed the `Home` link from the navbar, since we don't have a home/landing page.
* Refactored `alertToRender` function and reused it in `Login` and `Signup` components.

### Frontend testing
* Fixed tests by mocking the `useAuthorization` hook.
* Added a simple test that checks if `Dashboard` is rendering.

### CODE REVIEW

File: frontend/language/src/en.json [hunk 1]
```diff
@@ -25,7 +25,9 @@
   "header_form.factory.form.builder.link": "Form Builder",
   "header_form.factory.home.link": "Home",
   "header_form.factory.title.link": "Form Factory",
+  "login_page.auth.error": "You must be logged in to access the requested page.",
```
Comment (JamalAlabdullah) [inline frontend/language/src/en.json:28]: I think we can simplify the message to "Please log in, or create an account if you're new here."

File: frontend/language/src/en.json [hunk 2]
```diff
@@ -25,7 +25,9 @@
...
   "login_page.login.button": "Log in",
+  "login_page.login.error": "The server could not process your request. Please double check your credentials and try again.",
```
Comment (JamalAlabdullah) [inline frontend/language/src/en.json:30]: What about this "Could not sign you in, Please ensure your username and password are correct and try again."

File: frontend/main-app/src/components/Login/Login.tsx [hunk 3]
```diff
@@ -1,32 +1,35 @@
...
 
     if (formIsValid) {
       const apiUrl = getApiUrl();
       const targetUrl = `${apiUrl}/api/users/login`;
-      if (await axiosPostForm(targetUrl, formData)) {
+      if (await submitForm(targetUrl, formData)) {
         navigate("/dashboard");
       } else {
-        setServerError(t("signup_page.server.error"));
+        setErrorAlert("loginServerError");
```
Comment (JamalAlabdullah) [inline frontend/main-app/src/components/Login/Login.tsx:32]: What if we refactur handleSubmit to :

  ```
const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const formData = new FormData(event.currentTarget);
    const loginForm: LoginForm = Object.fromEntries(formData);
    const formIsValid: boolean = validateLoginForm({ loginForm, setFieldErrors });
    const apiUrl = getApiUrl();
    const targetUrl = `${apiUrl}/api/users/login`;
    try {
      const response = await submitForm(targetUrl, formData);
      if (response && formIsValid) navigate("/dashboard");
    }catch(error){
      setErrorAlert("loginServerError")
    }
  };
```

Comment (ErlingHauan) [reply frontend/main-app/src/components/Login/Login.tsx]: I see that we can reduce nesting, but the API request should not be made if the form is invalid.

What do you think about this?

```
const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    const formData = new FormData(event.currentTarget as HTMLFormElement);
    const loginForm: LoginForm = Object.fromEntries(formData);
    const formIsValid = validateLoginForm({ loginForm, setFieldErrors });

    if (!formIsValid) {
      return;
    }

    const apiUrl = getApiUrl();
    const targetUrl = `${apiUrl}/api/users/login`;

    try {
      await axios.post(targetUrl, loginForm, {
        withCredentials: true,
      });
      navigate("/dashboard");
    } catch (error) {
      console.log(error);
      setErrorAlert("loginServerError");
    }
  };
```

Comment (JamalAlabdullah) [reply frontend/main-app/src/components/Login/Login.tsx]: Looks good also😊

**Review by JamalAlabdullah — DISMISSED** [review]
Bra jobba😊, det er bare merg-conf som må fikses 😊

**Pull request merged**